metadata-models/src/main/pegasus/com/linkedin/dataset/Constraint.pdl

```diff
@@ -0,0 +1,34 @@
+namespace com.linkedin.dataset
+
+import com.linkedin.common.Urn
+
+/**
+ * Indicates a constraint that is applied to a dataset
+ */
+record Constraint {
+
+  /**
+  * The system in which the constraints are specified
+  */
+  // e.g. "greatexpectations"
+  constraintProvider: optional string
```

Comment:
Let's call this "platform" -- Usually we refer to external systems using DataPlatform Urns. You can see the DatasetKey.pdl file to see that. I think we should treat Great Expectation etc as a simple Data Platform Urn to keep things consistent.